### PR TITLE
Add delete autoscaling policy API

### PR DESCRIPTION
**Pull request by jasontedor**
This commit adds an API for deleting autoscaling policies.

### CODE REVIEW

**Comment by jasontedor** [comment]
@elasticmachine update branch

**Comment by jasontedor** [comment]
@elasticmachine update branch

File: x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/action/DeleteAutoscalingPolicyAction.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        @Override
+        public ActionRequestValidationException validate() {
+            return null;
+        }
+
+        @Override
+        public XContentBuilder toXContent(final XContentBuilder builder, final Params params) throws IOException {
+            builder.startObject();
+            {
+                builder.field("name", name);
```
Comment (henningandersen) [inline x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/action/DeleteAutoscalingPolicyAction.java:62]: I think the norm for Request `toXContent` is to not include fields that are not part of the REST body. For Instance, `PutAutoscalingPolicyAction.Request.toXContent` does not include the policy name. I tend to think this toXContent should generate an empty object or maybe the request should not implement `toXContentObject`?

**Review by henningandersen — APPROVED** [review]
LGTM.

File: x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/action/DeleteAutoscalingPolicyAction.java (hunk 1 above)
Comment (jasontedor) [reply x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/action/DeleteAutoscalingPolicyAction.java]: I opened #54643.

**Pull request merged**